src/passes/RemoveUnusedBrs.cpp

```diff
@@ -701,7 +702,51 @@ struct RemoveUnusedBrs : public WalkerPass<PostWalker<RemoveUnusedBrs>> {
     struct Optimizer : public PostWalker<Optimizer> {
       bool worked = false;
 
+      // Set of seen labels, used to generate a fresh label.
+      std::unordered_set<Name> labels;
+      // Used to generate fresh labels.
+      Index freshNameHint = 0;
+      // Whether we need to uniquify the labels at the end.
+      bool hasLabelConflict = false;
```

Comment:
Can we always uniquify at the end if `freshNameHint > 0` like in ReFinalize?

Reply:
Yes, but it would seem better to go the other direction and make ReFinalize more precisely detect conflicts like we do here.

Reply:
Why is that extra complexity worth it?

Reply:
In the vast majority of cases where there might potentially be a conflict, there will not actually be a conflict. Ideally we would run the extra uniquify pass when there actually is a conflict to fix. But maybe that's not worth the extra code complexity, since even cases where there might be a conflict should be rare. I'll simplify the code an add a TODO about the possible optimization.